### ISSUE TITLE
`help-jump-in-buffer` doesn't search through full heading text

### ISSUE THREAD

**Issue by tkossak** [syl20bnr/spacemacs#15014]
Hi, anybody knows how to make `SPC s j` (`lazy-helm/spacemacs/helm-jump-in-buffer`) show/search full org heading texts, not only the begining? It doesn't find any occurrences when I search for one word, but it does show the heading with this word, when I search for word from begining of the line. On the screenshot it seems like the heading is clipped (two dots ".." at the end):

![image](https://user-images.githubusercontent.com/3728245/130482148-59121217-0750-4fe2-8885-076356aea2a8.png)

After some time it does search full heading text but how to make it work properly all the time?

Thank you in advance for any help.

**Comment by lebensterben** [comment]
It is the intended behaviour to only search the headings.
What are you trying to do?

**Comment by tkossak** [comment]
Yes, this is what I'm trying to do - to search only headings (not their body) and it doesn't work, because some headings are clipped, as you see on the screenshot.

**Comment by lebensterben** [comment]
Please post some of the headings so I can try,

**Comment by tkossak** [comment]
Sample:

```
* ZGŁOSZENIA ================================================
** 50%KUP - info + dzieła * 50KUP * AEKK
```

Now I press `SPC s j` and I see that the second heading is clipped (as on the screenshot in first post). When I write "aekk" nothing is found.

**Comment by tkossak** [comment]
I made additional screenshots.

- This is how it looks when I open `tmp.org` and press `SPC s j` (notice clipped heading name):
![image](https://user-images.githubusercontent.com/3728245/130572454-554de5cc-42ed-4291-8bed-4e80c92d9d59.png)

- Now I write "aekk" and nothing is found:
![image](https://user-images.githubusercontent.com/3728245/130572675-92fe9b78-4ff2-4211-9d71-097572d6ab33.png)


- Then I modify the file by adding new empty line at the end, and spacemacs displays full heading name:
![image](https://user-images.githubusercontent.com/3728245/130572784-b9ff2f7d-5738-4469-b422-ff087c7efd04.png)

- and searches for it properly too:
![image](https://user-images.githubusercontent.com/3728245/130572862-72fd9106-c48f-47fb-8a99-f27849807706.png)

**Comment by lebensterben** [comment]
remove the trailing `=` from your headings

**Comment by tkossak** [comment]
Ok, I can shorten heading names, but I hope it's not final "solution", because I'd like to be able to search all my headings and not delete parts of them. Is it possible to search all my heading names in spacemacs? As I wrote it starts working after some time (maybe after I modify the buffer), so it seems more like a bug.

**Comment by lebensterben** [comment]
@tkossak 
`=` has special meaning in org mode syntax.

I guess it's working sometimes because the number of `=`. try adding a few more and it will break again.

**Comment by tkossak** [comment]
But the problem is not based on "=" char (I could live without "=" in headings perfectly fine) - I can insert whatever char, eg "a" and it breaks displaying again:

![image](https://user-images.githubusercontent.com/3728245/130581468-6aad6a54-19a3-42b0-8295-a7a05d9f82ed.png)


and searching too:

![image](https://user-images.githubusercontent.com/3728245/130581663-710b9795-836c-4014-9122-c2ba240021a1.png)

**Comment by lebensterben** [comment]
sorry you're right.

**Comment by lebensterben** [comment]
add
`(setq helm-org-outline-path nil)` in your `user-config`.

**Comment by tkossak** [comment]
Unfortunately it doesn't solve the problem. Worked only the first time that I restarted spacemacs, after following restarts still the same problem persists. My full user-config looks like this:

```lisp
(defun dotspacemacs/user-config ()
  "Configuration for user code:
This function is called at the very end of Spacemacs startup, after layer
configuration.
Put your configuration code here, except for variables that should be set
before packages are loaded."

  (setq helm-org-outline-path nil)
  (setq org-startup-indented 't)

  ; create zs/ze shortcuts:
  (defun hscroll-cursor-left ()
    (interactive "@")
    (set-window-hscroll (selected-window) (current-column)))
  (defun hscroll-cursor-right ()
    (interactive "@")
    (set-window-hscroll (selected-window) (- (current-column) (window-width) -1)))
  (define-key evil-normal-state-map "zs" 'hscroll-cursor-left)
  (define-key evil-normal-state-map "ze" 'hscroll-cursor-right)
  (setq auto-hscroll-mode 't)
  (setq hscroll-margin 0
        hscroll-step 1)

  ; vim additional insert mode exit keys:
  (key-chord-mode 1)
  (key-chord-define evil-insert-state-map "fj" 'evil-escape)
  )
```

and this is my full [.spacemacs](https://pastebin.com/gwzjdK4E).

**Comment by tkossak** [comment]
Maybe we should remove "not a bug" label? :)
